### PR TITLE
RUM-2778: Print error when no-op core is invoked

### PR DESCRIPTION
**Pull request by ambushwork**
### What does this PR do?

Print error when no-op core is invoked

### Motivation

RUM-2778

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](../CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/NoOpInternalSdkCore.kt [hunk 1]
```diff
@@ -121,9 +122,32 @@ internal object NoOpInternalSdkCore : InternalSdkCore {
...
+                REGISTER_FEATURE_INVOKED_ON_NO_OP_CORE_ERROR.format(
+                    Locale.US,
+                    feature.name
+                )
+            }
+        )
+    }
 
-    override fun getFeature(featureName: String): FeatureScope? = null
+    override fun getFeature(featureName: String): FeatureScope? {
```
Comment (0xnm) [inline dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/NoOpInternalSdkCore.kt:138]: This may get very spammy, given the number of calls to this method. Maybe we should print here only once?

Comment (ambushwork) [reply dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/NoOpInternalSdkCore.kt:138]: I think it depends, it's probably enough to print only once per feature, however if the client enables multiple features with no-op core and we print only the error for the first one, then it will be harder for them to discover the issue.

Comment (0xnm) [reply dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/NoOpInternalSdkCore.kt:138]: I mean we can leverage `onlyOnce` argument of the logger, it will be printed once per _each_ feature.

https://github.com/DataDog/dd-sdk-android/blob/e37f350c098dff484269e1679640523f563ee49b/dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/InternalLogger.kt#L111

**Review by hamorillo — DISMISSED** [review]
LGTM! 🚀

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/NoOpInternalSdkCore.kt (hunk 1 above)
Comment (ambushwork) [reply dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/NoOpInternalSdkCore.kt:138]: oh yes, I was thinking the same argument, but I didn't realize that the message is unique per feature, so it will print once per feature indeed.

Note: another change is made for the convenience of test for `onlyOnce`, and I don't see why we need to `get()` a new `SdkInternalLogger` every time.
```
// before
override val internalLogger: InternalLogger
        get() = SdkInternalLogger(this)

// now
override val internalLogger: InternalLogger = SdkInternalLogger(this)
```

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/NoOpInternalSdkCoreTest.kt [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        Datadog.setVerbosity(Int.MAX_VALUE)
+    }
+
+    @Test
+    fun `M not throw W registerFeature()`(@StringForgery fakeFeatureName: String) {
+        // Given
+        whenever(mockFeature.name) doReturn fakeFeatureName
+
+        // When + Then
+        Mockito.mockStatic(Log::class.java).use {
```
Comment (0xnm) [inline dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/NoOpInternalSdkCoreTest.kt]: Do we need to mock if we don't verify log calls anyway? Normally it should be just no-op without mocking.

Probably it is redundant here and in the 2 tests below.

**Review by 0xnm — DISMISSED** [review]

File: dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/NoOpInternalSdkCoreTest.kt (hunk 2 above)
Comment (ambushwork) [reply dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/NoOpInternalSdkCoreTest.kt]: remove mockStatic since it's not necessary, but keep the tests since they can still protect some corner cases.

**Review by 0xnm — APPROVED** [review]

**Review by hamorillo — APPROVED** [review]
